app/services/katello/event_daemon.rb

```diff
@@ -1,60 +1,91 @@
 module Katello
   class EventDaemon
-    def self.runnable?
-      if defined?(Rails::Console)
-        Rails.logger.info("Not starting Katello Event Daemon in the Rails Console")
-        return false
+    class << self
+      def initialize
+        FileUtils.touch(lock_file)
+        @cache = ActiveSupport::Cache::MemoryStore.new
       end
 
-      if $PROGRAM_NAME.match(/spring/)
-        Rails.logger.info("Spring is running in the environment; not starting Katello Event Daemon")
-        return false
+      def settings
+        SETTINGS[:katello][:event_daemon]
       end
 
-      if Rails.env.production? && $PROGRAM_NAME.match(/dynflow/)
-        Rails.logger.info("This appears to be the Dynflow process; not starting Katello Event Daemon")
-        return false
+      def pid
+        return unless pid_file && File.exist?(pid_file)
+
+        File.open(pid_file) { |f| f.read.to_i }
       end
 
-      if started?
-        Rails.logger.info("Katello Event Daemon already running; not attempting startup")
-        return false
+      def pid_file
+        pid_dir.join('katello_event_daemon.pid')
       end
 
-      !::Foreman.in_rake? && !Rails.env.test? && SETTINGS[:katello][:event_daemon_enabled]
-    end
+      def pid_dir
+        Rails.root.join('tmp', 'pids')
+      end
 
-    def self.start
-      if runnable?
-        services.values.each(&:run)
+      def lock_file
+        Rails.root.join('tmp', 'katello_event_daemon.lock')
```

Comment:
Not an issue for now, but if we try to stuff Foreman+Katello into containers, this path won't necessarily have to be shared between containers.

Reply:
Good point. By the time we get there I'd like to make this all safe to be running multiple instances of